File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>

using namespace std;
int n;
int main(){
  ios::sync_with_stdio(0);
  cin.tie(0);
  cin >> n;
  queue<int> q;
  while(n--){
    string word;
    cin >> word;
    if(word == "push"){
      int num;
      cin >> num;
      q.push(num);
    }
    else if(word == "pop"){
      if(q.empty()) cout << -1;
      else{
        cout << q.front();
        q.pop();
      }
      cout << "\n";
    }
    else if(word == "size") cout << q.size() << "\n";
    else if(word == "empty"){
      if(q.empty()) cout << 1;
      else cout << 0;
      cout << "\n";
    }
    else if(word == "front"){
      if(q.empty()) cout << -1;
      else cout << q.front();
      cout << "\n";
    }
    else{
      if(q.empty()) cout << -1;
      else cout << q.back();
      cout << "\n";
    }
  }
}

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Deque<Integer> q = new ArrayDeque<>();
        int n = Integer.parseInt(br.readLine());
        for(int i=0; i<n; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String com = st.nextToken();
            if(com.equals("push")){
                int num = Integer.parseInt(st.nextToken());
                q.add(num);
            }
            else if(com.equals("pop")){
                if(q.isEmpty()){
                    bw.write(-1+"\n");
                }
                else{
                    bw.write(q.poll()+"\n");
                }
            }
            else if(com.equals("size")){
                bw.write(q.size()+"\n");
            }
            else if(com.equals("empty")){
                if(q.isEmpty()){
                    bw.write(1+"\n");
                }
                else{
                    bw.write(0+"\n");
                }
            }
            else if(com.equals("front")){
                if(q.isEmpty()){
                    bw.write(-1+"\n");
                }
                else{
                    bw.write(q.peek()+"\n");
                }
            }
            else{
                if(q.isEmpty()){
                    bw.write(-1+"\n");
                }
                else{
                    bw.write(q.peekLast()+"\n");
                }
            }
        }
        bw.flush();
        bw.close();
    }
}